mainboilerplate/service.go

```diff
@@ -20,33 +14,25 @@ type ZoneConfig struct {
 // ServiceConfig represents identification and addressing configuration of the process.
 type ServiceConfig struct {
 	ZoneConfig
-	ID   string `long:"id" env:"ID" description:"Unique ID of this process. Auto-generated if not set"`
-	Host string `long:"host" env:"HOST" description:"Addressable, advertised hostname or IP of this process. Hostname is used if not set"`
-	Port string `long:"port" env:"PORT" description:"Service port for HTTP and gRPC requests. A random port is used if not set. Port may also take the form 'unix:///path/to/socket' to use a Unix Domain Socket"`
+	ID                string `long:"id" env:"ID" description:"Unique ID of this process. Auto-generated if not set"`
+	Host              string `long:"host" env:"HOST" description:"Addressable, advertised hostname or IP of this process. Hostname is used if not set"`
+	Port              string `long:"port" env:"PORT" description:"Service port for HTTP and gRPC requests. A random port is used if not set. Port may also take the form 'unix:///path/to/socket' to use a Unix Domain Socket"`
+	ServerCertFile    string `long:"server-cert-file" env:"SERVER_CERT_FILE" default:"" description:"Path to the server TLS certificate. This option toggles whether TLS is used. If absent, all other TLS settings are ignored."`
+	ServerCertKeyFile string `long:"server-cert-key-file" env:"SERVER_CERT_KEY_FILE" default:"" description:"Path to the server TLS private key"`
+	ServerCAFile      string `long:"server-ca-file" env:"SERVER_CA_FILE" default:"" description:"Path to the trusted CA for server verification of client certificates. When present, client certificates are required and verified against this CA. When absent, client certificates are not required but are verified against the system CA pool if presented."`
```

Comment:
If SERVER_CA_FILE is set, then clients MUST present a certificate and it must validate against SERVER_CA_FILE.

Otherwise, clients MAY present a certificate and if they do, it's verified.

Reply:
https://github.com/gazette/core/blob/johnny/auth/server/server.go#L186

Here's where this is toggled.

Reply:
I guess if you omit it, then it'd verify client certs using the system certs, since there's really no other sane default. And that seems like it's just the default behavior of Go, not something you setup explicitly.